File: python/3163.string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        prev = ""
        count = 0
        res = []

        for ch in word:
            if ch == prev and count < 9:
                count += 1
            else:
                if prev:
                    res.append(str(count))
                    res.append(prev)
                count = 1

            prev = ch

        if prev:
            res.append(str(count))
            res.append(prev)

        return "".join(res)

File: cpp/3163.string-compression-iii.cpp
#include <sstream>
#include <string>
using std::string;
class Solution {
public:
  string compressedString(string word) {
    char prev = 0;
    int counter = 0;
    std::stringstream res;

    for (char ch : word) {
      if (prev == ch && counter < 9) {
        counter++;
      } else {
        if (prev != 0) {
          res << counter;
          res << prev;
        }

        counter = 1;
      }

      prev = ch;
    }

    if (prev != 0) {
      res << counter;
      res << prev;
    }

    return res.str();
  }
};